src/main/kotlin/sliv/tool/importer/ProjectParser.kt

```diff
@@ -2,42 +2,77 @@ package sliv.tool.importer
 
 import javafx.collections.FXCollections
 import javafx.scene.control.TreeItem
+import javafx.scene.image.Image
+import javafx.scene.image.ImageView
+import sliv.tool.parsers.planes.ImagePlanesParser.extractUIDs
+import sliv.tool.project.model.*
 import java.io.File
+import kotlin.io.path.Path
 
 object ProjectParser {
-    const val IMAGE_DIRECTORY_NAME = "images"
+    private const val IMAGE_DIRECTORY_NAME = "images"
+    private val images = FXCollections.observableHashMap<Long, String>()
+    private val layers = FXCollections.observableArrayList<ProjectLayer>()
+    private val landmarks = FXCollections.observableHashMap<Long, MutableList<LandmarkFile>>()
+    private val frames = FXCollections.observableArrayList<ProjectFrame>()
 
-    val files = FXCollections.observableHashMap<Long, MutableList<OutputInfo>>()
-    val images = FXCollections.observableArrayList<ImageInfo>()
-
-    fun createTreeWithFiles(path: String?, tree: TreeItem<String>): TreeItem<String> {
+    fun parseDirectory(path: String): Project {
         if (path == null) {
-            return tree
+            throw Exception("Directory not selected")
         }
         var directory = File(path)
 
         for (folder in directory.listFiles()) {
-            if (folder.name == IMAGE_DIRECTORY_NAME) {
-                images.addAll(folder.listFiles().map {
-                    ImageInfo(it.nameWithoutExtension, it.absolutePath)
-                })
-            } else {
+            if (folder.name != IMAGE_DIRECTORY_NAME) {
+                var splitName = folder.name.split("_")
+                var name = splitName[0]
+                var kind = splitName[1]
```

Comment:
why not val too

Reply:
and what if name of the algorithm contains '_' by itself? I think, that kind should be the last part of the splitted, name is the all string before kind.
For example: 'fr1_desk_orb_keypoints'. 'fr1_desk_orb' is name',  'keypoints' is kind